inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_matmul_to_fc_or_gemm.cpp

```diff
@@ -213,10 +235,18 @@ ngraph::pass::ConvertMatMulToGemm::ConvertMatMulToGemm() {
         new_ops.push_back(gemm);
 
         if (gemm->get_shape() != output_shape) {
-            // This case is possible only when second input had exactly 1 dimension (that is not supported by GEMM operation)
-            // and for this case we have to reshape second input to first but this affects output shape (additional dimensions)
+            // This case is possible when one of the inputs has exactly 1 dimension (that is not supported by GEMM operation)
             // So to preserve output shape we insert additional reshape operation
-            auto reshape_output = op::util::reshapeTo(gemm, output_shape);
+            std::shared_ptr<ngraph::Node> reshape_output;
+            if (output_shape.size() == 0) {
+                std::vector<int64_t> dim_indices(gemm->get_shape().size());
+                std::iota(dim_indices.begin(), dim_indices.end(), 0);
+                reshape_output = std::make_shared<ngraph::opset1::Squeeze>(gemm,
```

Comment:
Do we have any functional tests that covers this branch?

Reply:
It was covered by [`backend/matmul.in.cpp`](https://github.com/openvinotoolkit/openvino/pull/4509/files#diff-62a71dd4b0e3eaeb1aef0482cf74b6ebc5db1e45d320a1fd8dd11ade7ff64a7d) tests (1D x 1D) 
so I've added corresponding test cases to the functional (single layer) tests 29eb4e5